### PR TITLE
Archive flags update [arflags]

### PR DESCRIPTION
**Pull request by camierjs**
Remove the 'u' flag in the ar command, to update all files in the archive.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2518](https://github.com/mfem/mfem/pull/2518) | @camierjs | @tzanio | @v-dobrev + @pazner | 09/07/21 | 09/18/21 | 09/19/21 | |

### CODE REVIEW

**Comment by camierjs** [comment]
LGTM

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by tzanio — APPROVED** [review]

**Review by pazner — APPROVED** [review]
This is great! Fixes the annoying `undefined reference` errors when doing incremental builds with nvcc.

File: CHANGELOG [hunk 1]
```diff
@@ -35,6 +35,8 @@ Version 4.3.1 (development)
 
 - Fixed several MinGW build issues on Windows.
 
+- Remove the 'u' flag in the ar command, to update all files in the archive.
```
Comment (pazner) [inline CHANGELOG:38]: Maybe explain why this change is made here?

Comment (camierjs) [reply CHANGELOG]: `done`, let me know if that is what you had in mind.

Comment (pazner) [reply CHANGELOG]: Looks good. Might be useful to mention that this issue comes up when compiling with nvcc. What do you think?

Comment (camierjs) [reply CHANGELOG]: isn't more related to the `ar` version on `linux` vs the `macos`, but depending on the compiler (`nvcc` is one)?

Comment (pazner) [reply CHANGELOG]: Ah, yeah, you're right. Ok, it's good as it is now.

**Comment by tzanio** [comment]
Merged in `next` for testing...
